website/src/components/Chat/ChatConfigSaver.tsx

```diff
@@ -4,14 +4,11 @@ import { ChatConfigFormData } from "src/types/Chat";
 import { setConfigCache } from "src/utils/chat";
 
 export const ChatConfigSaver = () => {
-  const { watch, formState } = useFormContext<ChatConfigFormData>();
+  const { watch } = useFormContext<ChatConfigFormData>();
   const config = watch();
   useEffect(() => {
-    // only update when form is changed
-    if (formState.isDirty) {
-      setConfigCache(config);
-    }
-  }, [config, formState.isDirty]);
+    setConfigCache(config);
+  }, [config]);
```

Comment:
when we change presets, we use the function `reset`, which sets the `isDirty` to false, so we are not saving.

Reply:
The isDirty check is require to avoid update on the first render which use the first model config from serverside, not the one from localstorage

Reply:
We should pass keepDirty option when update the preset:
```js
reset({ ...config, model_config_name: selectedModel }, {
     keepDirty: true,
});
```

Reply:
I refactored it, I added keep dirty, and switched to using `Controller` instead of `register` and it works now (hopefully)